datacite-findable-doi-handler/src/main/java/no/unit/nva/datacite/handlers/FindableDoiEventHandler.java

```diff
@@ -0,0 +1,112 @@
+import no.unit.nva.publication.doi.update.dto.DoiUpdateHolder;
+import nva.commons.utils.IoUtils;
+import nva.commons.utils.JacocoGenerated;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class FindableDoiEventHandler extends DestinationsEventBridgeEventHandler<PublicationHolder, DoiUpdateHolder> {
+
+    public static final String PUBLICATION_IS_MISSING_ERROR = "Publication is missing";
+    public static final String CUSTOMER_ID_IS_MISSING_ERROR = "CustomerId is missing";// log messages
+    public static final String RECEIVED_REQUEST_TO_MAKE_DOI_FINDABLE_LOG =
+        "Received request to set landing page (make findable) for DOI {} to landing page {} for {}";
+    static final String DOI_IS_MISSING_ERROR = "Doi is missing";
+    static final String PUBLICATION_ID_MISSING_ERROR = "Publication id is missing";
+    private static final Logger logger = LoggerFactory.getLogger(FindableDoiEventHandler.class);
+    private static final String SUCCESSFULLY_MADE_DOI_FINDABLE = "Successfully handled request for Doi {} : {}";
+    private final DoiClient doiClient;
+
+    @JacocoGenerated
+    public FindableDoiEventHandler() {
+        this(defaultDoiClient());
+    }
+
+    public FindableDoiEventHandler(DoiClient doiClient) {
+        super(PublicationHolder.class);
+        this.doiClient = doiClient;
+    }
+
+    @Override
+    protected DoiUpdateHolder processInputPayload(PublicationHolder input,
+                                                  AwsEventBridgeEvent<AwsEventBridgeDetail<PublicationHolder>> event,
+                                                  Context context) {
+
+        Publication publication = getPublication(input);
+        URI customerId = getCustomerId(publication);
+        URI doiUri = getDoi(publication);
+
+        URI publicationId = getPublicationId(publication);
+        URI landingPage = getLandingPage(publicationId);
+        logger.debug(RECEIVED_REQUEST_TO_MAKE_DOI_FINDABLE_LOG, doiUri, landingPage, customerId);
+
+        try {
+            Doi doi = null; // Use PR for Doi.
+            doiClient.setLandingPage(customerId, doi, landingPage);
+            DoiUpdateHolder doiUpdateHolder = new DoiUpdateHolder("doi.updateDoiStatus",
+                new Builder()
+                    .withPublicationId(publicationId)
+                    .withModifiedDate(Instant.now())
+                    .withDoi(doiUri).build());
+            logger.debug(SUCCESSFULLY_MADE_DOI_FINDABLE, doiUri, doiUpdateHolder.toJsonString());
+            return doiUpdateHolder;
+
+        } catch (ClientException e) {
+            throw new ClientRuntimeException(e);
+        }
+    }
+
+    private static DoiClient defaultDoiClient() {
+        String dataCiteConfigJson = AppEnv.getDataCiteConfig();
+        DataCiteConfigurationFactory dataCiteConfigurationFactory = new DataCiteConfigurationFactory(
+            IoUtils.stringToStream(dataCiteConfigJson));
```

Comment:
The DataciteConfig string should be static (static variables should be used when possible for better performance for Lambdas). 
Also we have a string turned to an InputStream turned to a String again for no reason.

Reply:
This was fixed in https://github.com/BIBSYSDEV/nva-doi-registrar-client/pull/29 .

Reply:
Pushed update to use the new constructor.

But `AppEnv.getDataCiteConfig();` is static, so is the `DoiClient`. Isn't this enough or?